### DISCUSSION TITLE
Conditional formatting in result set

### DISCUSSION THREAD

**Discussion by hbsts** [microsoft-search/pnp-modern-search#3313] [Q&A] [answered]
Hello everyone,

In the result set from the search engine there is a column status (resp. managed property) in my content type. Depending on the value (open, delayed) of the managed property status, I want to set the background of the column or the entire row of the result set to green resp. red (like formatting views in a SharePoint list).

Is this possible using the PnP Search Results WebPart?

Thanks,
Christian

**Comment by kasperbolarsen** [comment]
Hi 
You absolutely can. In both the Handlebar templates and the Adaptive Cards you will have that option.

see https://www.youtube.com/watch?v=3QrAcFgYmro as a primer as you will need some conditional logic and some CSS to archive the goal.

**Reply by hbsts** [reply]
Hi Kasper,

Thanks a lot for the video. It gave me some new insights and helped me with another issue. 

But to colour the value or the column is still unclear to me. Let us say the column is named {{slot item @root.slots.Status}}. If the value is "delayed", "delayed" should be in red color in the webpart. That would be easiest, I guess.

Is it the Column Value {{slot item @root.slots.Department}} under Manage Columns where some if..then logic has to be added? Do you know any examples in the PnP Search documentation may be how this can be achieved? 

Best, Christian

**Reply by kasperbolarsen** [reply, marked as answer]
Take a look at https://microsoft-search.github.io/pnp-modern-search/usage/search-results/slots/
Which explains the use of slots 
In your case  {{slot item @root.slots.Department}} , the slot Department might be matched to a managed property by the same name, but you will have to open "Layout slots" to check.
You could try to browse some of the existing templates or https://dotnetgeekster.wordpress.com/2022/11/16/customizing-the-pnp-modern-search-using-handlebar-helpers/
To see how to react to an given condition/value

**Reply by hbsts** [reply]
Amazing, it works. Tak!
I was able to got it working for the Cards template. But I need to use Details List. With Details List template it is not possible to update the output like that, right? 
Anyway, great WebPart and thanks again for your support.

**Reply by kasperbolarsen** [reply]
All of the Handlebar templates can be customized using that approach .
Getting it to look fancy is beyond my capabilities,  but ask a front end dev friend of yours and everything is possible 😁

**Reply by hbsts** [reply]
Oh yes, you are right again. It is working for Details List, too. Great!
For Cards template I was updating the Results Template directly. For Details Lists, I added this code to Manage Columns section to color the text red. 
{{#eq (slot item @root.slots.Department) "Department123"}}                                           
	<span style="color:red;">{{slot item @root.slots.Department}}</span>
{{else}}
	{{slot item @root.slots.Department}}
{{/eq}}

**Reply by kasperbolarsen** [reply]
You might find this useful when tinkering with templates,  https://ms365thinking.blogspot.com/2023/10/work-with-your-handlebar-based-layouts.html?m=1
